### ISSUE TITLE
String formatting ignores arguments that aren't ObjC objects

### ISSUE THREAD

**Issue by charlieMonroe** [drkameleon/iswift#83]
Great 1.9 release! The improvements save me a lot of manual fix-up. Thanks! Will be posting a few more issues, though :)

Example:

```
[NSString stringWithFormat:@"%li", time(NULL)]
```

Result:

```
"%li"
```

Should be:

```
"\(time(nil))"
```

**Comment by drkameleon** [comment]
> Great 1.9 release! The improvements save me a lot of manual fix-up. Thanks!

Glad to hear that.

> Will be posting a few more issues, though :)

Sure do! :)

---

Now, as for the issue it self, the string formatter now roughly supports the following:

```coffeescript
/%(h|hh|l|ll|q|L|z|t|j)?[@dDuUxXoOfeEgGcCsSpaAF]/
```

Which means: `%li` is not included.

So, I'll add it. :)

**Comment by charlieMonroe** [comment]
Perhaps there should be some warning if there is an unknown percentage sequence (e.g. `%0.2f`), since this is very error prone and not everyone needs to notice this. The only allowed unhandled percentage sequence should be `%%`...

**Comment by drkameleon** [comment]
I think you may be right... Let me run a few tests and chances are I'll do it in a more... universal way. ;-)

**Comment by drkameleon** [comment]
Issue fixed as of the upcoming 2.0.
